src/components/RecipeDetailMain.js

```diff
@@ -0,0 +1,141 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+
+class RecipeDetailMain extends Component {
+  constructor() {
+    super();
+
+    this.main = this.main.bind(this);
+  }
+
+  main(recipeDetail) {
+    const { history: { location: { pathname } } } = this.props;
+
+    let strThumb = '';
+    let str = '';
+    let category = '';
+
+    const { strMealThumb, strDrinkThumb, strDrink, strMeal,
+      strCategory, strAlcoholic, strInstructions,
+      strYoutube } = recipeDetail;
+
+    if (pathname.includes('comidas')) {
+      strThumb = strMealThumb;
+      str = strMeal;
+      category = strCategory;
+    } else {
+      strThumb = strDrinkThumb;
+      str = strDrink;
+      category = strAlcoholic;
+    }
```

Comment:
O que acha de simplificar um pouco a complexidade dessa parte? 😉 
```suggestion
    const { strMealThumb, strDrinkThumb, strDrink, strMeal,
      strCategory, strAlcoholic, strInstructions,
      strYoutube } = recipeDetail;

    let strThumb = strMealThumb || strDrinkThumb;
    let str = strMeal || strDrink;
    let category = strCategory || strAlcoholic;
```

Reply:
Boaaaa. 
Vou apenas inverter as variáveis das linhas 20 e 18 logo também.